ClimsoftVer4/ClimsoftVer4/frmNewFormDaily2.vb

```diff
@@ -0,0 +1,103 @@
+﻿
+Public Class frmNewFormDaily2
+    Private bFirstLoad As Boolean = True
+
+    Private Sub frmNewFormDaily2_Load(sender As Object, e As EventArgs) Handles MyBase.Load
+
+        If bFirstLoad Then
+            InitaliseDialog()
+        End If
+    End Sub
+
+    Private Sub InitaliseDialog()
+        Dim dtbVis, dtbCld, dtbPrec, dtbTemp As New DataTable
+
+        ucrFormDaily.setYearAndMonthLink(ucrYearSelector, ucrMonth)
+        AssignLinkToKeyField(ucrFormDaily)
+
+        ucrVisibilityUnits.SetField("visUnits")
+        ucrVisibilityUnits.SetTableName("form_daily2")
+        ucrVisibilityUnits.bFillFromDataBase = False
+        dtbVis.Columns.Add("visUnits", GetType(String))
+        dtbVis.Rows.Add("metres")
+        dtbVis.Rows.Add("yards")
+        ucrVisibilityUnits.SetPossibleValues(dtbVis)
```

Comment:
Patrick what are you suggesting? This would be nice if the SetPossibleValues had a version which took in the column name and type as well as a parameter array for the values. Which would essentially push these 5 lines of code down nicely into the SetPossibleValues routine.

Reply:
That's even a much better suggestion.